oslabs-beta/DockerLocal
# Fetches to Github v4 API for user's repositories

## Description:
- Added fetches to get all user repositories from github's v4 GraphQL API
## Changes I Made:
In `AddRepos.tsx` :
- Takes a **user's login name** and **encrypted access token** from **cookies**
→ Stores the username for use in the fetches through template literal queries
→ Decrypts the access token places it in the fetch header template literal

- Fetches for...
→  A user's **public repositories**
→  Repositories the user has **collaborated** on
→  Repositories from **organizations** the user is a member of

## How to Test:
- With the running electron app, after signing in and clicking the 'Add Repositories' button in the developer (toggle) menu, a developer can see the fetched (and parsed) repositories in the chrome developer console.

## Issues:
- Neither the v3 or v4 API allowed access to a **user's private repositories**, so **only public repositories are returned**
- **Possible authentication issue** with passing in the access token through the described header
